### PR TITLE
Merge commit '05ee88915520d1dd82da94a016a9374a1f3a8129' from upstream

### PR DESCRIPTION
**Pull request by timsifive**

### CODE REVIEW

**Review by timsifive — COMMENTED** [review]
Lots of changes, but not much that could actually affect us.

Most changes that can be ignored are:
* Add angie debug device.
* PLD changes.
* JTAG driver changes, which seem like trivial error handling.
* Lots of target changes changing to LOG_TARGET* macros, many of which convert a warning to an error.
* Lots of ESP32 changes, none of which address the RISC-V core present on esp32s2 and esp32s3 (IIRC)
* Lots of `pld device` -> `pld create` in config files

**Comment by timsifive** [comment]
As usual, there are minor checkpatch complaints that I will override.

**Comment by aap-sc** [comment]
@timsifive , just a note. Once this commit is merged it would not hurt to address relevant TODO from this MR: https://github.com/riscv/riscv-openocd/pull/897/files

File: tcl/target/esp_common.cfg [hunk 1]
```diff
@@ -77,12 +77,21 @@ proc create_esp_target { ARCH } {
 	set_esp_common_variables
 	create_esp_jtag
 	create_openocd_targets
+<<<<<<< HEAD
 	configure_openocd_events $ARCH
+=======
+	configure_openocd_events
+>>>>>>> 05ee88915520d1dd82da94a016a9374a1f3a8129
```
Comment (MarekVCodasip) [inline tcl/target/esp_common.cfg]: There are several unresolved conflicts in this file. I believe the HEAD variant should be taken and the other one dropped.

**Comment by MarekVCodasip** [comment]
We took a look with @JanMatCodasip. It all looks fine, except for the esp_common.cfg file mentioned above.

**Comment by timsifive** [comment]
I did my best to resolve the remaining esp_common.cfg conflicts. I don't have a target to test on, however.
@erhankur can you confirm that the ESP config still works?

**Comment by erhankur** [comment]
> I did my best to resolve the remaining esp_common.cfg conflicts. I don't have a target to test on, however. @erhankur can you confirm that the ESP config still works?

@timsifive LGTM. It works. Thanks

File: tcl/target/esp_common.cfg (hunk 1 above)
Comment (JanMatCodasip) [reply tcl/target/esp_common.cfg]: Resolved by Tim (below).

**Review by JanMatCodasip — APPROVED** [review]

**Pull request merged**